### PR TITLE
app-misc/khard: Bump to 0.18.0, bump to EAPI="8"

### PR DESCRIPTION
**Pull request by holgersson32644**
Please note that upstream maintenance changed since last release.

Signed-off-by: Nils Freydank <holgersson@posteo.de>

### CODE REVIEW

File: app-misc/khard/khard-0.18.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="8"
```
Comment (winterheart) [inline app-misc/khard/khard-0.18.0.ebuild:4]: You may just use EAPI=8 as per recommendations of [DevManual](https://devmanual.gentoo.org/ebuild-writing/eapi/index.html)

File: app-misc/khard/khard-0.18.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		dev-python/sphinx-autoapi
+		dev-python/sphinx-autodoc-typehints
+	)
+"
+
+DOCS=( CHANGES CONTRIBUTING.rst README.md doc/source/examples/khard.conf.example )
+
+distutils_enable_tests setup.py
+
+src_compile() {
```
Comment (winterheart) [inline app-misc/khard/khard-0.18.0.ebuild:52]: You redefining src_compile() just for doc generation. Simplier way is use `python_compile_all()` as per https://projects.gentoo.org/python/guide/distutils.html#python-compile

File: app-misc/khard/khard-0.18.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	dev-python/ruamel-yaml[${PYTHON_USEDEP}]
+	dev-python/ruamel-yaml-clib[${PYTHON_USEDEP}]
+	dev-python/unidecode[${PYTHON_USEDEP}]
+	dev-python/vobject[${PYTHON_USEDEP}]
+"
+BDEPEND="
+	test? (
+		dev-python/setuptools_scm[${PYTHON_USEDEP}]
+	)
+	doc? (
```
Comment (winterheart) [inline app-misc/khard/khard-0.18.0.ebuild:41]: Distutils-r1 provides more convenient approach for generating python's packages doc generations like `distutils_enable_sphinx doc`.
See https://projects.gentoo.org/python/guide/distutils.html#building-documentation-via-sphinx.

File: app-misc/khard/khard-0.18.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+src_compile() {
+	distutils-r1_src_compile
+
+	if use doc; then
+		emake -j1 -C doc/ html text man info
+	fi
+}
+
+src_install() {
```
Comment (winterheart) [inline app-misc/khard/khard-0.18.0.ebuild:60]: Same here, better to use `python_install_all()` as per https://projects.gentoo.org/python/guide/distutils.html#python-install

File: app-misc/khard/khard-0.18.0.ebuild (hunk 3 above)
Comment (holgersson32644) [reply app-misc/khard/khard-0.18.0.ebuild:44]: well, with all the fiddling with custom additions I actually don't find that one more convenient, but I'll try it anyways.

File: app-misc/khard/khard-0.18.0.ebuild (hunk 2 above)
Comment (holgersson32644) [reply app-misc/khard/khard-0.18.0.ebuild]: Pretty cool, thanks for the hint - I totally missed that one.

File: app-misc/khard/khard-0.18.0.ebuild (hunk 1 above)
Comment (holgersson32644) [reply app-misc/khard/khard-0.18.0.ebuild]: Habits die hard, I changed it locally. Actually I prefer to see that EAPIs are actually strings, but as the rest of the Gentoo world abuses them as numbers anyways there is no real benefit.

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**